src/main/java/UI/ViewLikedUsers.java

```diff
@@ -0,0 +1,179 @@
+package UI;
+
+import entities.UserAccount;
+import interfaceAdapters.UserHistoryPresenter;
+import useCase.GenerateUserHistory;
+
+import javax.swing.*;
+import java.awt.*;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+
+/**
+ * UI for the liked users page.
+ */
+public class ViewLikedUsers extends JFrame implements ActionListener {
+
+    // Get current user // Test user for now
+    public UserAccount user1 = new UserAccount("johnd", "John Doe", 20, "her", "USA",
+            "ILL", "CHI", "M", "H","Watching", "123");
+    //UserAccount currentUser;
+
+    GenerateUserHistory generateHistory = new GenerateUserHistory();
+
+    // Get the total number of likes this user has
+    String totalLikes = Integer.toString(generateHistory.getTotalLiked(user1));
+    String likedAge = Integer.toString(generateHistory.getLikedAgeRange(user1));
+    String likedCity = generateHistory.getLikedCity(user1);
+    String likedCountry = generateHistory.getLikedCountry(user1);
+    String likedGender = generateHistory.getLikedGender(user1);
+    String likedInterest = generateHistory.getLikedInterest(user1);
+
+    // New instance of JFrame
+    JFrame frame = new JFrame();
+
+    // Buttons
+    JButton backButton = new JButton("Return ⏎");
+    JButton blockButton = new JButton("Block ✖");
+    JButton unlikeButton = new JButton("Unlike ☹");
+
+    // Labels
+    JLabel likedTitle = new JLabel("LIKED USERS");
+
+    // Statistics Labels
+    JLabel totalLikedUsersTitle = new JLabel("Total Number of Liked Users: ");
+    JLabel likedAgeTitle = new JLabel("Most Common Age of Liked Users: ");
+    JLabel likedCityTitle = new JLabel("Most Common City of Liked Users: ");
+    JLabel likedCountryTitle = new JLabel("Most Common Country: ");
+    JLabel likedGenderTitle = new JLabel("Most common Gender of Liked Users: ");
+    JLabel likedInterestTitle = new JLabel("Most common Interests: ");
+    JLabel totalLikesUsersNumber = new JLabel(totalLikes);
+    JLabel likedAgeNumber = new JLabel(likedAge);
+    JLabel likedCityString = new JLabel(likedCity);
+    JLabel likedCountryString = new JLabel(likedCountry);
+    JLabel likedGenderString = new JLabel(likedGender);
+    JLabel likedInterestString = new JLabel(likedInterest);
+
+    // List of users they have liked
+    String[] likedUsers = generateHistory.likedUsernames(user1);
+    JComboBox<String> likedBox = new JComboBox<>(likedUsers);
+
+
+    public ViewLikedUsers(){
+
+        // Add frames
+        frame.add(likedTitle);
+        frame.add(backButton);
+        frame.add(likedBox);
+        frame.add(blockButton);
+        frame.add(unlikeButton);
+        frame.add(totalLikedUsersTitle);
+        frame.add(likedAgeTitle);
+        frame.add(likedCityTitle);
```

Comment:
You may want to put this into some sort of builder method or put it in some sort of method since this looks potentially smelly and really long.